File: app/src/main/java/com/example/cxh/kotlinstudy/kotlin/kt09_singleton/java/LazyThreadSafeSynchronized.java
package com.example.cxh.kotlinstudy.kotlin.kt09_singleton.java;

// 单例2
public class LazyThreadSafeSynchronized {
    private static LazyThreadSafeSynchronized INSTANCE;

    private LazyThreadSafeSynchronized(){}

    public static synchronized LazyThreadSafeSynchronized getInstance(){
        if(INSTANCE == null){
            INSTANCE = new LazyThreadSafeSynchronized();
        }
        return INSTANCE;
    }
}

File: app/src/main/java/com/example/cxh/kotlinstudy/kotlin/kt09_singleton/kotlin/LazyThreadSafeSynchronized.kt
package com.example.cxh.kotlinstudy.kotlin.kt09_singleton.kotlin


class LazyThreadSafeSynchronized private constructor() {
    companion object {
        private var instance: LazyThreadSafeSynchronized? = null

        @Synchronized
        fun get(): LazyThreadSafeSynchronized {
            if (instance == null) instance = LazyThreadSafeSynchronized()
            return instance!!
        }
    }

}